### PR TITLE
Setup the website to be deployed in docker

### PR DESCRIPTION
**Pull request by fozziethebeat**
This prepares the website for deployment in docker and demonstrates an end to end docker compose setup.

This includes a handful of related changes:
*  Fixing some npm build and warn issues
*  Fixing a next-js build configuration issue related to docker
*  Replacing a few UI elements with chakra-ui components where appropriate
*  Adding a dockerfile for the website
*  Adding an end to end docker-compose config and README

### CODE REVIEW

File: website/src/pages/grading/grade-output.tsx [hunk 1]
```diff
@@ -134,7 +134,7 @@ const GradeOutput = () => {
           <button
             type="button"
             onClick={() => submitResponse(tasks[0])}
-            className="nline-flex items-center rounded-md border border-transparent bg-indigo-600 px-4 py-2 text-sm font-medium text-white shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
+            className="inline-flex items-center rounded-md border border-transparent bg-indigo-600 px-4 py-2 text-sm font-medium text-white shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
```
Comment (andreaskoepf) [inline website/src/pages/grading/grade-output.tsx:137]: nice catch! :+1:

**Review by andreaskoepf — APPROVED** [review]
lgtm, good work! :-)

**Review by yk — APPROVED** [review]
lgtm, thank you

File: website/src/pages/grading/grade-output.tsx (hunk 1 above)
Comment (fozziethebeat) [reply website/src/pages/grading/grade-output.tsx:137]: That was Yannic who caught that :)

**Review by k-nearest-neighbor — APPROVED** [review]
🚢

**Pull request merged**